pkg/protocol/http1/client.go

```diff
@@ -594,6 +620,11 @@ func (c *HostClient) acquireConn() (cc *clientConn, err error) {
 			}
 		}()
 
+		// Note: In the case of setting MaxConnWaitTimeout, if the number
+		// of connections in the connection pool exceeds the maximum
+		// number of connections and needs to establish a connection while
+		// waiting, the dialtimeout on the hostclient is used instead of
+		// the dialtimeout in request options.
```

Comment:
also add explanation on the commentary of client.ClientOptions.DialTimeout field